File: Beginner/Java/String/IsAnagram.java
class Solution {
    public boolean isAnagram(String s, String t) {
        if (s.length() != t.length())
            return false;
        int[] sc = new int[26];
        int[] st = new int[26];
        for (int i = 0; i < s.length(); i ++) {
            sc[s.charAt(i) - 'a'] ++;
            st[t.charAt(i) - 'a'] ++;
        }
        for (int i = 0; i < 26; i ++)
            if(sc[i] != st[i])
                return false;
        return true;
    }
}

File: Beginner/C/String/IsAnagram.c
bool isAnagram(char* s, char* t) {
    int cs[26];
    int ct[26];
    for(int i = 0; i < 26; i ++) {
        cs[i] = 0; ct[i] = 0;
    }
    while(s && t && *s != 0 && *t != 0) {
        cs[((int)*s) - 97] ++;
        ct[((int)*t) - 97] ++;
        s ++; t ++;
    }
    if (*s != 0 || *t != 0)
        return false;
    for(int i = 0; i < 26; i ++) {
        if (cs[i] != ct[i])
            return false;
    }
    return true;
}
